webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/zarr3/Zarr3Array.scala

```diff
@@ -1,47 +1,37 @@
 package com.scalableminds.webknossos.datastore.datareaders.zarr3
 
+import com.scalableminds.util.tools.{Fox, JsonHelper}
 import com.scalableminds.util.cache.AlfuCache
-import com.scalableminds.util.tools.Fox
 import ucar.ma2.{Array => MultiArray}
 import com.scalableminds.webknossos.datastore.datareaders.{AxisOrder, ChunkReader, ChunkUtils, DatasetArray}
 import com.scalableminds.webknossos.datastore.datavault.VaultPath
 import com.scalableminds.webknossos.datastore.models.datasource.DataSourceId
 import com.typesafe.scalalogging.LazyLogging
-import play.api.libs.json.{JsError, JsSuccess, Json}
-
-import java.io.IOException
-import java.nio.charset.StandardCharsets
+import com.scalableminds.util.tools.Fox.box2Fox
 import scala.collection.immutable.NumericRange
 import scala.concurrent.ExecutionContext
 
 object Zarr3Array extends LazyLogging {
-  @throws[IOException]
+
   def open(path: VaultPath,
            dataSourceId: DataSourceId,
            layerName: String,
            axisOrderOpt: Option[AxisOrder],
            channelIndex: Option[Int],
-           sharedChunkContentsCache: AlfuCache[String, MultiArray]): Zarr3Array = {
-    val headerBytes = (path / Zarr3ArrayHeader.ZARR_JSON).readBytes()
-    if (headerBytes.isEmpty)
-      throw new IOException("'" + Zarr3ArrayHeader.ZARR_JSON + "' expected but is not readable or missing in store.")
-    val headerString = new String(headerBytes.get, StandardCharsets.UTF_8)
-    val header: Zarr3ArrayHeader =
-      Json.parse(headerString).validate[Zarr3ArrayHeader] match {
-        case JsSuccess(parsedHeader, _) =>
-          parsedHeader
-        case errors: JsError =>
-          throw new Exception("Validating json as zarr v3 header failed: " + JsError.toJson(errors).toString())
-      }
-    new Zarr3Array(path,
-                   dataSourceId,
-                   layerName,
-                   header,
-                   axisOrderOpt.getOrElse(AxisOrder.asCxyzFromRank(header.rank)),
-                   channelIndex,
-                   sharedChunkContentsCache)
-  }
-
+           sharedChunkContentsCache: AlfuCache[String, MultiArray])(implicit ec: ExecutionContext): Fox[Zarr3Array] =
+    for {
+      headerBytes <- (path / Zarr3ArrayHeader.ZARR_JSON)
+        .readBytes() ?~> s"Could not read header ${Zarr3ArrayHeader.ZARR_JSON}"
+      header <- JsonHelper.parseAndValidateJson[Zarr3ArrayHeader](headerBytes) ?~> "Could not parse array header"
+      _ <- DatasetArray.assertChunkSizeLimit(header.bytesPerChunk)
```

Comment:
```suggestion
```
Can not be used here, since with the way sharding is implemented, zarr3 chunks can be way bigger (and still used, since the second chunking happens with the sharding codec).